src/Data/List/NonEmpty/Properties.agda

```diff
@@ -49,6 +58,46 @@ toList->>= : ∀ {ℓ} {A B : Set ℓ}
              (List⁺.toList (xs >>= f))
 toList->>= f (x ∷ xs) = begin
   List.concat (List.map (List⁺.toList ∘ f) (x ∷ xs))
-    ≡⟨ cong List.concat $ map-compose {g = List⁺.toList} (x ∷ xs) ⟩
+    ≡⟨ cong List.concat $ Listₚ.map-compose {g = List⁺.toList} (x ∷ xs) ⟩
   List.concat (List.map List⁺.toList (List.map f (x ∷ xs)))
     ∎
+
+length-++⁺ : (l : List A) (l′ : List⁺ A) → length (l ++⁺ l′) ≡ List.length l + length l′
+length-++⁺ [] l′          = refl
+length-++⁺ (x ∷ l) l′
+  rewrite length-++⁺ l l′ = refl
+
+length-++⁺′ : (l : List A) (l′ : List⁺ A) → length (l ++⁺ l′) ≡ suc (List.length l + List.length (List⁺.tail l′))
```

Comment:
the ' is not very indicative here. I think `tail` should appear in the name of the property. Plus, it also feels like this ought to be derived, i.e. the proof should combine a property of `length` and one of `+`.

Reply:
I will change the name. however, proving it using other theorems is rather cumbersome. this lemma can be proved using the previous one, and then reason `a + (1 + b) = (1 + a + b)`  while using induction it's pretty much the same proof.